[RISCV] LMUL lists for indexed and strided loads

ppenzin opened the pull request:
Create additional lists representing valid LMULs for strided and indexed load of particular element sizes.

ppenzin committed:
[RISCV] LMUL lists for indexed and strided loads

Create additional lists representing valid LMULs for strided and indexed load
of particular element sizes.

wangpc-pp reviewed (approved):
LGTM.

wangpc-pp commented on llvm/lib/Target/RISCV/RISCVSchedSiFive7.td:
```diff
@@ -582,10 +582,8 @@ multiclass SiFive7WriteResBase<int VLEN,
       defm : LMULWriteResMX<"WriteVSTOX8", [VCQ, VS], mx, IsWorstCase>;
     }
   }
-  // TODO: The MxLists need to be filtered by EEW. We only need to support
```
Do we need to keep this TODO?

ppenzin replied:
I can revert that. If we try to implement this we would need to create a macro that takes EEW and returns a MxList, and this change doesn't quite get there. @mshockwave any objections on keeping the TODO?

mshockwave replied:
I think this comment might be outdated already: the fact that it excludes MF8 means that it already accounts for EEW, as explained by the comment itself.

ppenzin commented on llvm/lib/Target/RISCV/RISCVScheduleV.td:
```diff
@@ -19,6 +19,14 @@ defvar SchedMxListFW = !listremove(SchedMxList, ["M8", "MF8"]);
 // Used for widening floating-point Reduction as it doesn't contain MF8.
 defvar SchedMxListFWRed = SchedMxListF;
+// Used for indexed and strided loads of 8 bit lanes, same as full MX list
+defvar SchedMxListDS8 = SchedMxList;
```
Focused on wrong common letter, sorry

topperc replied:
This is just a list of allowed MX values for a particular EEW. It doesn't need to mention strided or indexed at all. Maybe SchedMxListEEW8, SchedMxListEEW16, SchedMxListEEW32, SchedMxListEEW64?

ppenzin replied:
Addressed.

ppenzin committed:
DS -> SX in MxLists aliases

S and UX/OX are the common parts of SchedWrite names for strided and segmented
operations, hence "SX" suffix, followed by EEW.

ppenzin committed:
Use EEW in MxList aliases

topperc reviewed (approved):
LGTM

mshockwave reviewed (approved):
LGTM
